packages/backend/src/database/migrations/20230116111752_add_pinned_list_table.ts

```diff
@@ -0,0 +1,73 @@
+import { Knex } from 'knex';
+
+const PinnedListTableName = 'pinned_list';
+const PinnedItemsTableName = 'pinned_items';
+
+export async function up(knex: Knex): Promise<void> {
+    await knex.schema.alterTable('projects', (table) => {
+        table.unique(['project_uuid']);
+    });
+    await knex.schema.alterTable('saved_queries', (table) => {
+        table.unique(['saved_query_uuid']);
+    });
+    await knex.schema.alterTable('dashboards', (table) => {
+        table.unique(['dashboard_uuid']);
+    });
+    if (!(await knex.schema.hasTable(PinnedListTableName))) {
+        await knex.schema.createTable(PinnedListTableName, (table) => {
+            table
+                .uuid('pinned_list_uuid')
+                .primary()
+                .notNullable()
+                .defaultTo(knex.raw('uuid_generate_v4()'));
+            table
+                .uuid('project_uuid')
```

Comment:
don't we need to create an `unique` rule for project_uuid ? can we have more than 1 pinned list per project ?

Reply:
I think we want that option open for the future.
eg: each user has a pinned list for the same project, a list for each space in the project.

Reply:
So these are kind of like "collections" a project can have multiple collections and those collections can contain charts/dashboards. It seems very similar to a space but more simple

Reply:
I'm gonna make this table specific ( project_pinned_list )  which can only be 1 per project so we can add the restriction. And once we have a list per space we can have a separate table `space_pinned_list`.